web/src/main/java/org/springframework/security/web/server/authentication/DelegatingServerAuthenticationConverter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.authentication;
+
+import java.util.List;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.util.Assert;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * A {@link ServerAuthenticationConverter} that delegates to other
+ * {@link ServerAuthenticationConverter} instances.
+ *
+ * @author DingHao
+ * @since 6.3
+ */
+public class DelegatingServerAuthenticationConverter implements ServerAuthenticationConverter {
+
+	private final List<ServerAuthenticationConverter> delegates;
+
+	public DelegatingServerAuthenticationConverter(ServerAuthenticationConverter... converters) {
+		this(List.of(converters));
+	}
+
+	public DelegatingServerAuthenticationConverter(List<ServerAuthenticationConverter> converters) {
+		Assert.notEmpty(converters, "converters cannot be null");
+		this.delegates = converters;
+	}
+
+	@Override
+	public Mono<Authentication> convert(ServerWebExchange exchange) {
+		return Flux.fromIterable(this.delegates).concatMap((converter) -> converter.convert(exchange)).next();
```

Comment:
If the first delegate throws an error, the thread will be terminated. DelegatingServerAuthenticationConverter must iterates all delegates. It seems that you need to use concatMapDelayError.

Reply:
Done